web/yo/app/scripts/controllers/drugs.js

```diff
@@ -42,7 +42,11 @@ angular.module('oncokbApp')
 
 
             function hasSameName(newDrugName, uuid) {
-                return _.some(mainUtils.getKeysWithoutFirebasePrefix($scope.drugList), (key) => ($scope.drugList[key].uuid !== uuid && (newDrugName === $scope.drugList[key].drugName || $scope.drugList[key].synonyms !== undefined && $scope.drugList[key].synonyms.indexOf(newDrugName) > -1)) === true);
+                return _.some(mainUtils.getKeysWithoutFirebasePrefix($scope.drugList), function(key){
+                    if(($scope.drugList[key].uuid !== uuid && (newDrugName === $scope.drugList[key].drugName || $scope.drugList[key].synonyms !== undefined && $scope.drugList[key].synonyms.indexOf(newDrugName) > -1)) === true){
```

Comment:
What if it returns `undefined`? What will happen in curation platform?

Reply:
It will be handled in $scope.saveDrugName. undefined means can't find any.

Reply:
`Undefined` is quite easy to cause errors. Could you just return an empty string?
```
return $scope.drugList[key].uuid !== uuid && (newDrugName === $scope.drugList[key].drugName || $scope.drugList[key].synonyms !== undefined && $scope.drugList[key].synonyms.indexOf(newDrugName) > -1) ? key : '';
```

Reply:
Sorry, I didn't explain it clearly.  The result returned will be the result of _.some function. According to the Lodash document, "Returns true if any element passes the predicate check, else false".